### PR TITLE
[RFC] Use manager plugin to get commands and features

### PR DESCRIPTION
**Pull request by aschempp**
As discussed with @leofeyer, the Contao Manager API should use *features* instead of a version number. This allows the Contao Manager to dynamically enable or disable UI options based on the supported features, which is way easier than testing for a number.

This PR allows Contao Manager Plugins to register API commands and announce features. This is more of a preparation because the Contao Manager obviously must know about a feature to actually use it, but at least this can be used by the `contao/manager-bundle` to enable application configuration (like `prepend_locale`).

/cc @contao/developers

### CODE REVIEW

**Comment by leofeyer** [comment]
I like this a lot. Good job @aschempp!

**Comment by aschempp** [comment]
Update for latest plugin. The tests are failing because plugin PR is not merged and released yet.

File: src/Api/Application.php [hunk 1]
```diff
@@ -97,13 +134,18 @@ protected function getDefaultInputDefinition(): InputDefinition
     protected function getDefaultCommands(): array
     {
         $commands = parent::getDefaultCommands();
+        $commands[] = new VersionCommand($this);
```
Comment (leofeyer) [inline src/Api/Application.php:145]: Why is this not added dynamically?

Comment (aschempp) [reply src/Api/Application.php:145]: What do you mean by dynamically? This is the basic command that always must be there to determine what the API is capable of.

Comment (leofeyer) [reply src/Api/Application.php:145]: Well, not quite. The API version will never change again, so …

Comment (aschempp) [reply src/Api/Application.php:145]: So? The command is still relevant for everything to get features etc. 
Also, the version _might_ change if we come up with something new, like UI for manager configuration.

**Review by leofeyer — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @aschempp. I like this a lot.

**Comment by Toflar** [comment]
Yes, really nice work!

**Pull request merged**